### PR TITLE
Augment the Python Bindings to allow using precompiled Generators

### PR DESCRIPTION
**Pull request by steven-johnson**
Augment the Python Bindings to allow using precompiled Generators (written in C++) at JIT time via the Python bindings, mimicking the C++ stub functionality.

### CODE REVIEW

File: python_bindings/stub/PyStub.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
+// Small stub to produce a module-specific entry point
+// for a Python module. It assumes/requires that it is
+// linked with PyStubImpl.o to be useful.
+//
+// Note that this quite deliberately
+// doesn't include any PyBind11 headers; it is designed to
+// be compiled without PyBind11 being available at compile time,
+// to simplify build requirements some downstream environments.
```
Comment (abadams) [inline python_bindings/stub/PyStub.cpp]: requirements some -> requirements in some

File: src/Generator.h [hunk 2]
```diff
@@ -2733,6 +2733,7 @@ class GeneratorBase : public NamesInterface, public GeneratorContext {
     Func get_first_output();
     Func get_output(const std::string &n);
     std::vector<Func> get_output_vector(const std::string &n);
+    std::vector<std::vector<Func>> get_output_vector();
```
Comment (abadams) [inline src/Generator.h]: Took me a long time to figure out why this is a ```vector<vector<..>>``` These methods could use some brief descriptive comments.

**Comment by steven-johnson** [comment]
https://github.com/halide/Halide/pull/2720 needs to land before this one can.

**Comment by steven-johnson** [comment]
Monday morning review ping.

**Comment by abadams** [comment]
I added a pair of comments last week, LGTM apart from those.

**Comment by jrk** [comment]
I skimmed but am too tied up for a deep review (and I also think the Python world is much less mission-critical at this point, so features trump rigor to a much greater degree than in the core). I'll just give a professor-level response: cool! I want this! 👍 → 🚢/🐑

**Comment by steven-johnson** [comment]
Pending clean build, this is ready to land; I have two LGTMs, but no approvals, so someone, please step up :-)

**Pull request merged**